app/post_test.go

```diff
@@ -2767,6 +2769,32 @@ func TestGetPostIfAuthorized(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestGetPostIfAuthorizedNew(t *testing.T) {
```

Comment:
This is a verbatim copy of the previous test. We need to refactor much more here.